### DISCUSSION TITLE
Serverless deployments

### DISCUSSION THREAD

**Discussion by olivermrbl** [medusajs/medusa#2853] [Feature Requests]
### What
Support for running Medusa in a serverless environment

**Comment by amkayondo** [comment]
What platforms will be supported?

**Comment by hamburger-fries** [comment]
I want to see the server running via Vercel + S3 + Upstash + Supabase.

**Reply by agustif** [reply]
That was certainly foreseeing as the new product modules is indeed using the new Vercel KV storage powered by Upstash ;)

**Comment by ask007learning** [comment]
If you have a website with low to moderate traffic, using AWS Lambda to run it would essentially result in no cost.

**Comment by tgdn** [comment]
Any news about this?

**Comment by kb-0912** [comment]
Any update?

**Comment by olivermrbl** [comment]
Hi everyone. 

An update from our side. 

Yesterday, we launched a beta version of our Product Module, which can run in any Node.js compatible runtime, including serverless environments such as Vercel Functions. 

You can read the product announcement [here](https://medusajs.com/blog/announcing-product-module/) or alternative dive right into the docs on how to use it in Next.js [here](https://docs.medusajs.com/modules/products/serverless-module). 

This marks an important milestone for us, as we now have a proof-of-concept as well as the foundation in place to further modularize our core engine. Eventually, all commerce modules will be capable of running in serverless environments - our roadmap in the near future is centered around realizing exactly this.

**Comment by ensidialabs** [comment]
@olivermrbl, please explain why you chose to use MicroORM, and not, for example, Prisma or Drizzle ORM? Thank you and all team MedusaJS.

**Reply by olivermrbl** [reply]
I have tried to flesh out the reasoning in https://github.com/medusajs/medusa/discussions/4431 :)

**Comment by cirolosapio** [comment]
the discussion is closed so I assume that is possible to deploy a medusa server with serverless(?)
